agent/handlers/v3/task_metadata_handler.go

```diff
@@ -53,11 +54,24 @@ func TaskMetadataHandler(state dockerstate.TaskEngineState, ecsClient api.ECSCli
 
 		taskResponse, err := v2.NewTaskResponse(taskARN, state, ecsClient, cluster, az, containerInstanceArn, propagateTags, false)
 		if err != nil {
-			errResponseJSON, err := json.Marshal("Unable to generate metadata for task: '" + taskARN + "'")
+			statusCode := http.StatusInternalServerError
+			errorResponse := fmt.Sprintf("Unable to generate metadata for v3 task: '%s'.", taskARN)
+
+			if awsErr, ok := err.(awserr.Error); ok {
+				if reqErr, ok := err.(awserr.RequestFailure); ok {
+					statusCode = reqErr.StatusCode()
+				}
+				errorLogMessage := fmt.Sprintf(" Received the error code: %s, the error message: %s,"+
+					" and the status code: %d for v3 task '%s'",
```

Comment:
could you change this so the error message changes from 
```
"Unable to generate metadata for v3 task: 'taskArn'. Received the error code: ThrottlingException, the error message: Rate exceeded, and the status code: 400 for v3 task 'taskArn'"
```
to 
```
"Unable to generate metadata for v3 task: 'taskArn'. Error code: 400 - ThrottlingException: Rate exceeded"
```

Reply:
same comments for v4 as well.

Reply:
Updated the message as suggested, as shown [here](https://github.com/aws/amazon-ecs-agent/blob/991ff6eb2bab4386e43c762e5b331725c90ebc5d/agent/handlers/v4/task_metadata_handler.go#L60-L61). Thanks!